utensor_cgen/transformer/ns_transformer.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding:utf8 -*-
+r"""Namescope Transformer
+
+Transformers that get rid of namescope/nodes which are not needed 
```

Comment:
Well, I think my comment here is not accurate in deed.
An `NamescopeTransformer` is just transformer which knowns how to deal with `tf.namescope` things.
It can remove it, replace it with other operations,...etc.
They are still transformers and in my definition for transformer, it is an object which has a method `transform` which will take a `uTensorGraph` and return a new `uTensorGraph`.

I think what you propose is more appropriate for an optimizer, which is also an graph transformer.

Reply:
cool beans